lib/main.dart

```diff
@@ -607,6 +649,236 @@ class _GamePageState extends State<GamePage> {
       ),
     );
   }
+
+  void _save() {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return BoilerplateDialog(
+          title: 'Save to',
+          children: [
+            ...Directory.current
+                .listSync()
+                .whereType<File>()
+                .where((e) => e.path.endsWith('.save'))
+                .map((e) => Row(
+                      mainAxisSize: MainAxisSize.min,
+                      children: [
+                        TextButton(
+                            onPressed: () {
+                              _saveToFile(e.path);
+                              Navigator.pop(context);
+                            },
+                            child: Text(
+                                '${e.path.substring(e.path.lastIndexOf('/') + 1, e.path.length - 5)}')),
+                        TextButton(
+                          child: Icon(Icons.delete),
+                          onPressed: () {
+                            _deleteFile(e.path);
+                          },
+                        ),
+                      ],
+                    )),
+            TextButton(
+              child: Text('New Save'),
+              onPressed: () {
+                _newSave();
+              },
+            ),
+            TextButton(
+              child: Text('Cancel'),
+              onPressed: () {
+                Navigator.pop(context);
+              },
+            )
+          ],
+        );
+      },
+    );
+  }
+
+  // TODO: make deleting update the list in the load save dialog (same for saving)
+  void _load() {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return BoilerplateDialog(title: 'Load Save', children: [
```

Comment:
this uses a fixed list of children, which will be very slow when you open a directory with millions of files